### PR TITLE
Add Typescript types for builds functions

### PR DESCRIPTION
**Pull request by timneutkens**

### CODE REVIEW

File: packages/next/build/index.js [hunk 1]
```diff
@@ -1,50 +1,34 @@
...
-  }
+  await runCompiler(configs)
+  await writeBuildId(distDir, buildId)
 }
 
-function runCompiler (compiler) {
+function runCompiler (config) {
   return new Promise(async (resolve, reject) => {
-    const webpackCompiler = await webpack(await compiler)
+    const webpackCompiler = await webpack(config)
```
Comment (quietshu) [inline packages/next/build/index.js]: Why’s `await` removed here?

Comment (timneutkens) [reply packages/next/build/index.js]: We already await configs

Comment (quietshu) [reply packages/next/build/index.js]: so it used to have double `await`s 😅

**Review by shuding — APPROVED** [review]

File: packages/next/build/index.js (hunk 1 above)
Comment (timneutkens) [reply packages/next/build/index.js]: Correct, hence why I removed it

**Pull request merged**